src/bin/redact.js

```diff
@@ -22,24 +23,6 @@ if (helpFlag) {
 
 ioUtils.readFromFileOrStdin(argv._[0])
   .then(ioUtils.parseAsSerialized)
-  .then(redact)
+  .then(recursivelyProcessAll.bind(null, parser.sourceToRedacted.bind(parser)))
```

Comment:
It took me a little while to wrap my head around this, and part of that challenge was mentally checking how the output of `parseAsSerialized` maps to the parameters of `recursivelyProcessAll`.  I know it would denormalize a little bit, but I think this might read a let better if you kept your `redact` helper around, just implemented it in terms of `recursivelyProcessAll`.

```js
ioUtils.readFromFileOrStdin(argv._[0])
  .then(ioUtils.parseAsSerialized)
  .then(redact)
  .then(ioUtils.formatAsSerialized)
  .then(ioUtils.writeToFileOrStdout.bind(ioUtils, argv.o));

function redact(data) {
  return recursivelyProcessAll(parser.sourceToRedacted.bind(parser), data);
}
```

It's more vebose but it would have been more obvious, for example, that the `isValue` argument is intentionally omitted from the first `recursivelyProcessAll` call.

Reply:
sure thing!